### PR TITLE
Message Broadcasting System

### PR DESCRIPTION
**Pull request by Hithroc**
Well. I implemented a message broadcasting system.
It uses Google Form responses spreadsheet as a mailing list. The spreadsheet must have the following format:
l

This will be the default format if you just create a Google Form with only one field "Username" in it.

**Notice!** "daily_water_thread.py" modifications: I added so each time the script is executed it also sends a message with the link to the thread.
### How to set it up

You need to install gspread and oauth2client python packages:

```
pip3 install --user gspread
pip3 install --user oauth2client
```

Then you need to get Google API Signed Credentials. You can read about it [here](http://gspread.readthedocs.io/en/latest/oauth2.html) (steps from 1 to 3). Save the JSON file as `google.json` in the bot's directory.

Then you need to create a Google Form and a responses spreadsheet for it. Then you should share the spreadsheet with the e-mail you will find in `google.json` file.

You also need to place the spreadsheet's ID (you can find it in URL) in the 'config.py'

Example of an ID in an URL:
     spreadsheets/d/**12NlnL1Dfbwr60tm1xFQKmj0-Zh45UuG6KDZRtMs6ohk**/edit
### Unsubscribing

Each time you execute "broadcast" function from the `send_mails.py` module, it will check for unread messages in the inbox with "!unsubscribe" message body. Then it will remove the user who sent the message from the list. And only after that it will send the message to everyone in the list (new list).

### CODE REVIEW

**Comment by Hithroc** [comment]
Bonus :)
l

File: send_mails.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# Checks if there are any people that want to unsubscribe and remove them
+# from the list
+def check_for_unsubscribes(wks):
+    # Get unread inbox messages
+    g = r.get_unread()
+    for message in g:
+        # If someone wants to unsubscribe
+        if message.body == "!unsubscribe":
+            # delete them from the list
+            gs.unsub(wks, str(message.author.name))
```
Comment (Hithroc) [inline send_mails.py:15]: This `str` is completely unnecessary I just forgot to remove it.

**Comment by t-jayw** [comment]
Hey, sorry I haven't had time to get to this yet; it's been a busy week. 

Will have time over this weekend!

**Comment by Hithroc** [comment]
The code here is can no longer be quick merged, thus I'm closing the pull request.

**Pull request closed without merging**